build.gradle.kts

```diff
@@ -0,0 +1,143 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+    id("com.github.johnrengelman.shadow") version "5.2.0" apply false
+}
+
+// Expose DiSCo version to subprojects
+val discoVersion by extra("0.10.0")
+
+subprojects {
+    version = "2.7.0"
+    group = "com.amazonaws"
+
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+
+    // Configure the shadow jar task, which does shading, to run after Gradle runs the jar task
+    pluginManager.withPlugin("com.github.johnrengelman.shadow") {
+        tasks {
+            named<DefaultTask>("assemble") {
+                dependsOn(named<ShadowJar>("shadowJar"))
+            }
+            named<ShadowJar>("shadowJar") {
+                // Suppress the "-all" suffix on the jar name, simply replace the default built jar instead
+                archiveClassifier.set(null as String?)
+
+                dependsOn(named<Jar>("jar"))
```

Comment:
Do we need to even run jar task? For reference, here's my pattern https://github.com/anuraaga/aws-opentelemetry-java-instrumentation/blob/master/otelagent/build.gradle.kts#L62

Reply:
I like your way better :) will change.